shop.py

```diff
@@ -39,3 +39,5 @@ def tryNbuy(customerMoney, customerRequest, purchaseAttempts):
 
 
 tryNbuy(customerMoney1, customerRequest1, purchaseAttempts1)
+
+# comment
```

Comment:
Good attempt, expected to see the use of classes and functions for more efficiency.

Upon testing, the first error recieved is a KeyError: when an item is selected in either complete lower case or upper case. This should be handled. 

Upon selecting the exit option, it does throw an AssertionError.
Good to see that adding to the current balance and updating the balance until it is sufficient enough for the purchase is good.

Expected to see the program raise a ValueError after an invalid input of a maximum of 3 attempts. The current program kicks out after the first wrong input.

This program can be refactored to use functions for blocks of the program such as greeting, purchase, Maximum attempts. e.tc.

The program should also improve upon the use of comments to explain eaxh line/ block of code as much as possible.